Index: src/obstacle/examples/wscript
===================================================================
new file mode 100644
--- /dev/null
+++ b/src/obstacle/examples/wscript
@@ -0,0 +1,6 @@
+# -*- Mode: python; py-indent-offset: 4; indent-tabs-mode: nil; coding: utf-8; -*-
+
+def build(bld):
+    obj = bld.create_ns3_program('obstacle-example', ['obstacle'])
+    obj.source = 'obstacle-example.cc'
+
