Index: src/mobility/helper/ns2-mobility-helper.h
===================================================================
--- a/src/mobility/helper/ns2-mobility-helper.h
+++ b/src/mobility/helper/ns2-mobility-helper.h
@@ -19,7 +19,6 @@
  * Author: Mathieu Lacage <mathieu.lacage@sophia.inria.fr>
  * Contributors: Thomas Waldecker <twaldecker@rocketmail.com>
  *               Martín Giachino <martin.giachino@gmail.com>
- * Modified by:  Marco Romanelli <marco.romanelli.1@studenti.unipd.it>
  */
 #ifndef NS2_MOBILITY_HELPER_H
 #define NS2_MOBILITY_HELPER_H
@@ -36,7 +35,7 @@ class ConstantVelocityMobilityModel;
 /**
  * \ingroup mobility
  * \brief Helper class which can read ns-2 movement files and configure nodes mobility.
- *
+ * 
  * This implementation is based on the ns2 movement documentation of ns2
  * as described in http://www.isi.edu/nsnam/ns/doc/node172.html
  *
@@ -66,7 +65,7 @@ class ConstantVelocityMobilityModel;
  * The following tools are known to support this format:
  *  - BonnMotion http://net.cs.uni-bonn.de/wg/cs/applications/bonnmotion/
  *  - SUMO http://sourceforge.net/apps/mediawiki/sumo/index.php?title=Main_Page
- *  - TraNS http://trans.epfl.ch/
+ *  - TraNS http://trans.epfl.ch/ 
  *
  *  See usage example in examples/mobility/ns2-mobility-trace.cc
  *
@@ -104,12 +103,6 @@ public:
    */
   template <typename T>
   void Install (T begin, T end) const;
-
-	/**
-	 * Prevent nodes from moving; model will set only first position of each node
-	 */
-	void DisableNodeMovements (void);
-
 private:
   /**
    * \brief a class to hold input objects internally
@@ -137,8 +130,7 @@ public:
    * \return pointer to a ConstantVelocityMobilityModel
    */
   Ptr<ConstantVelocityMobilityModel> GetMobilityModel (std::string idString, const ObjectStore &store) const;
-  std::string m_filename; //!< filename of file containing ns-2 mobility trace
-	bool m_movementsEnabled;
+  std::string m_filename; //!< filename of file containing ns-2 mobility trace 
 };
 
 } // namespace ns3
@@ -146,7 +138,7 @@ public:
 namespace ns3 {
 
 template <typename T>
-void
+void 
 Ns2MobilityHelper::Install (T begin, T end) const
 {
   class MyObjectStore : public ObjectStore
